src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -463,10 +466,36 @@ export class PageControllerBase {
 
       viewModel.backLink = this.getBackLink(progress)
 
+      const missingEmailWarning = await this.buildMissingEmailWarningModel(
+        request,
+        isStartPage
+      )
+
+      if (missingEmailWarning) {
+        return h.view(this.viewName, { ...viewModel, ...missingEmailWarning })
+      }
+
       return h.view(this.viewName, viewModel)
     }
   }
 
+  async buildMissingEmailWarningModel(request: Request, isStartPage?: boolean) {
+    const { params } = request
+    // Warn the user if the form has no notification email set only on start page and summary page
+    if (isStartPage || params.path === 'summary') {
+      const { slug } = params as { slug: string }
+      const { notificationEmail } = await getFormMetadata(slug)
```

Comment:
Just linking to https://github.com/DEFRA/forms-runner/pull/484#discussion_r1808802631 because the code has moved

```suggestion
      const { notificationEmail } = await getFormMetadata(params.slug)
```

Reply:
I am also using slug in line 491. I need it in a const.

Reply:
done